pkg/crlprovider/crl_fetcher.go

```diff
@@ -0,0 +1,75 @@
+/*
+Copyright The Ratify Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package crlprovider
+
+import (
+	"crypto/x509"
+	"fmt"
+
+	corecrl "github.com/notaryproject/notation-core-go/revocation/crl"
+)
+
+// maxCRLSize is the maximum size of CRL in bytes
+//
+// The 32 MiB limit is based on investigation that even the largest CRLs
+// are less than 16 MiB. The limit is set to 32 MiB to prevent
+const maxCRLSize = 32 * 1024 * 1024 // 32 MiB
+
+// BytesFetcher implements the Fetcher interface to fetch CRL from the given bytes
+type BytesFetcher struct {
+	// Cache stores fetched CRLs and reuses them until the CRL reaches the
+	// NextUpdate time.
+	// If Cache is nil, no cache is used.
+	Cache corecrl.Cache
+
+	// DiscardCacheError specifies whether to discard any error on cache.
+	//
+	// ErrCacheMiss is not considered as an failure and will not be returned as
+	// an error if DiscardCacheError is false.
+	DiscardCacheError bool
+}
+
+// NewBytesFetcher creates a new BytesFetcher with the given HTTP client
+func NewBytesFetcher() (*BytesFetcher, error) {
```

Comment:
What's the usage of `error` input param in this function?

Reply:
I am about to discuss with our PM about the config in network isolation scenario further update this constructor,